radixdlt/src/main/java/org/radix/GenerateUniverses.java

```diff
@@ -164,6 +164,8 @@ private Atom createGenesisAtom(byte magic, long timestamp, long planck) throws E
 		);
 		genesisAtom.sign(universeKey);
 
+		LedgerAtom ledgerAtom = LedgerAtom.convertFromApiAtom(genesisAtom);
```

Comment:
We could just avoid the conversion and coupling by throwing a different exception.  Not really sure that ConstraintMachineValidationException is the right thing for this largely stand-alone piece of code that is unrelated to the constraint machine.

Reply:
That's true but I think this will get resolved once we formalize atom construction in the RE.